api/spotguide.go

```diff
@@ -0,0 +1,87 @@
+package api
+
+import (
+	"net/http"
+	"strings"
+
+	"github.com/banzaicloud/pipeline/auth"
+	pkgCommon "github.com/banzaicloud/pipeline/pkg/common"
+	"github.com/banzaicloud/pipeline/spotguide"
+	"github.com/gin-gonic/gin"
+	"github.com/jinzhu/gorm"
+)
+
+// GetSpotguide get detailed information about a spotguide
+func GetSpotguide(c *gin.Context) {
+	spotguideName := strings.TrimPrefix(c.Param("name"), "/")
```

Comment:
Why is `name` path param optional? If no spot guide `name` is passed in than the consequent `spotguide.GetSpotguide(spotguideName)` call will return not found as spotguideName is an empty string.

Reply:
`name` is not optional, I just had to use `*` because otherwise, I can't capture the spotguide's name which can contain `/` (because it is the name of the github repo it is originating from). But the `if == ""` check is missing you are right!